tools/codegen/codespec/config.go

```diff
@@ -43,7 +89,21 @@ var transformations = []AttributeTransformation{
 	createOnlyTransformation,
 }
 
+var dataSourceTransformations = []AttributeTransformation{
+	aliasTransformation,
+	overridesTransformation,
+	// Note: createOnlyTransformation is excluded for data sources (read-only, no create operation)
+}
+
 func applyAttributeTransformations(schemaOptions config.SchemaOptions, attributes *Attributes, parentPaths *attrPaths) error {
+	return applyAttributeTransformationsList(schemaOptions, attributes, parentPaths, transformations)
+}
+
+func applyDataSourceAttributeTransformations(schemaOptions config.SchemaOptions, attributes *Attributes, parentPaths *attrPaths) error {
+	return applyAttributeTransformationsList(schemaOptions, attributes, parentPaths, dataSourceTransformations)
+}
```

Comment:
nit: we can remove this function and simply call `applyAttributeTransformations` right?

Reply:
I see, makes sense